network/manager.go

```diff
@@ -732,8 +734,11 @@ func (nm *networkManager) GetNumberOfEndpoints(ifName string, networkId string)
 }
 
 // GetEndpointID returns a unique endpoint ID based on the CNI mode.
-func (nm *networkManager) GetEndpointID(containerID, ifName string) string {
+func (nm *networkManager) GetEndpointID(containerID, ifName string, nicType cns.NICType) string {
 	if nm.IsStatelessCNIMode() {
+		if nicType == cns.DelegatedVMNIC {
```

Comment:
Can you explain the reasoning for adding this? What does it fix and what baout other nic types (Frontend, etc.). How does it affect windows and linux (since I believe there are delegated vmnics in both)